File: AirPurrr/app/src/main/java/com/example/trubul/airpurrr/DataReader.java
package com.example.trubul.airpurrr;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class DataReader {

    String getData(InputStreamReader streamReader) {
        BufferedReader in = null;
        String inputLine;

        try {
            in = new BufferedReader(streamReader);
            StringBuilder stringBuilder = new StringBuilder();

            //Check if the line we are reading is not null
            while ((inputLine = in.readLine()) != null) {
                stringBuilder.append(inputLine).append('\n');
            }

            return stringBuilder.toString();
        }
        catch (IOException e) {
            e.printStackTrace();
        }
        finally {
            if (in != null) {
                try {
                    in.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }
        return null;
    }
}

File: AirPurrr/app/src/main/java/com/example/trubul/airpurrr/DataReader.kt
package com.example.trubul.airpurrr

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader

internal class DataReader {

    fun getData(streamReader: InputStreamReader): String? {
        var reader: BufferedReader? = null
        var inputLine: String? = null

        try {
            reader = BufferedReader(streamReader)
            val stringBuilder = StringBuilder()

            //Check if the line we are reading is not null
            while ({ inputLine = reader.readLine(); inputLine }() != null) {
                stringBuilder.append(inputLine).append('\n')
            }

            return stringBuilder.toString()
        } catch (e: IOException) {
            e.printStackTrace()
        } finally {
            if (reader != null) {
                try {
                    reader.close()
                } catch (e: IOException) {
                    e.printStackTrace()
                }

            }
        }
        return null
    }
}
